lib/rules/no-wait-for-side-effects.ts

```diff
@@ -48,22 +56,92 @@ export default createTestingLibraryRule<Options, MessageIds>({
       );
     }
 
+    function isRenderInVariableDeclaration(node: TSESTree.Node) {
+      return (
+        isVariableDeclaration(node) &&
+        node.declarations.some((declaration) =>
+          helpers.isRenderVariableDeclarator(declaration)
+        )
+      );
+    }
+
+    function isRenderInAssignment(node: TSESTree.Node) {
+      if (
+        !isExpressionStatement(node) ||
+        !isAssignmentExpression(node.expression)
+      ) {
+        return false;
+      }
+
+      const expressionIdentifier = getPropertyIdentifierNode(
+        node.expression.right
+      );
+
+      if (!expressionIdentifier) {
+        return false;
+      }
+
+      return helpers.isRenderUtil(expressionIdentifier);
+    }
+
+    function isRenderInImplicitReturnAssignment(node: TSESTree.Node) {
+      if (!isAssignmentExpression(node)) {
+        return false;
+      }
+
+      const expressionIdentifier = getPropertyIdentifierNode(node.right);
+
+      if (!expressionIdentifier) {
+        return false;
+      }
+
+      return helpers.isRenderUtil(expressionIdentifier);
+    }
+
+    function isExpressionRenderUtil(expression: TSESTree.Expression) {
+      if (!isAssignmentExpression(expression)) {
+        return false;
+      }
+
+      const expressionIdentifier = getPropertyIdentifierNode(expression.right);
+
+      if (!expressionIdentifier) {
+        return false;
+      }
+
+      return helpers.isRenderUtil(expressionIdentifier);
+    }
```

Comment:
As these 2 functions are doing exactly the same and only having the argument type differently, I would merge them